### PR TITLE
Add upgrade handler to add default navs for markers

### PR DESCRIPTION
**Pull request by nullpointer0x00**
## Description

Add an upgrade handler to set the navs of Figure markers values.  This file contains all the added navs: https://github.com/provenance-io/provenance/blob/64e9cb994fb0d950c7b9a5c5ff32d8155c875cd6/app/navs_mainnet.go

It was merged to main on accident.  Therefore, it is not showing up in the diff.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by Taztingo — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * Bump cometbft to v0.34.29 (from v0.34.28) [PR 1649](https://github.com/provenance-io/provenance/pull/1649).
 * Add genesis/init for Marker module send deny list addresses. [#1660](https://github.com/provenance-io/provenance/issues/1660)
 * Add automatic changelog entries for dependabot. [#1674](https://github.com/provenance-io/provenance/issues/1674)
+* Add upgrade handler to set net assert values to markers [PR 1712](https://github.com/provenance-io/provenance/pull/1712).
```
Comment (iramiller) [inline CHANGELOG.md:73]:
```suggestion
* Add upgrade handler to set net asset values to markers [PR 1712](https://github.com/provenance-io/provenance/pull/1712).
```

File: app/upgrades.go [hunk 2]
```diff
@@ -346,6 +349,33 @@ func updateMaxSupply(ctx sdk.Context, app *App) {
...
+		var hasNav bool
+		err := app.MarkerKeeper.IterateNetAssetValues(ctx, record.GetAddress(), func(nav markertypes.NetAssetValue) bool {
+			hasNav = true
+			return true
+		})
+		if err != nil {
+			ctx.Logger().Error(fmt.Sprintf("unable iterate net asset values for marker %v: %v", record, err))
+			return false
+		}
+		if !hasNav {
```
Comment (iramiller) [inline app/upgrades.go:368]: I don't think this is a good default ... we could set a default for HASH on testnet to a NAV of 15¢ (1:1500000000) as a good example of what we expect a token with a 1 billion base unit to have against a cent denom.

File: CHANGELOG.md [hunk 3]
```diff
@@ -70,6 +70,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
 * Ensure IBC marker has matching supply [#1706](https://github.com/provenance-io/provenance/issues/1706).
 ### Bug Fixes
```
Comment (SpicyLemon) [inline CHANGELOG.md:75]: Could you throw a space above this in this PR too (while fixing the other changelog comment), please?

Comment (iramiller) [reply CHANGELOG.md:75]: space was added in the other changelog PR ...

File: app/upgrades.go (hunk 2 above)
Comment (iramiller) [reply app/upgrades.go:387]: We should delete this `!hasNav` section unless we have a default we use for new markers which we could set here to match...

File: app/upgrades.go [hunk 4]
```diff
@@ -114,6 +115,15 @@ var upgrades = map[string]appUpgrade{
...
 			updateMaxSupply(ctx, app)
+
+			denomToNav := map[string]markertypes.NetAssetValue{
+				"usd.deposit": markertypes.NewNetAssetValue(sdk.NewInt64Coin(markertypes.UsdDenom, 1000), 1),
+				"tp13gc5xe8375msm7y6jhf752e5hcrjypmcpf9wldsjprhnpzlkhqaqawxujf.investment": markertypes.NewNetAssetValue(sdk.NewInt64Coin(markertypes.UsdDenom, 10), 1),
+				"tp1cxuqqyjjf5x66jvlmtvj3juppn370ev7rr5cja3ml65nzhxgvpkszfuvtw.investment": markertypes.NewNetAssetValue(sdk.NewInt64Coin(markertypes.UsdDenom, 10), 33600000),
+				"pm.sale.pool.7v2gsuvnudyfvuig50r3k3":                                      markertypes.NewNetAssetValue(sdk.NewInt64Coin(markertypes.UsdDenom, 19026640), 1),
+			}
+			addMarkerNavs(ctx, app, denomToNav, markertypes.NewNetAssetValue(sdk.NewInt64Coin(markertypes.UsdDenom, int64(150)), 1))
+
```
Comment (SpicyLemon) [inline app/upgrades.go:126]: `saffron-rc1` has already been run on testnet. This will need to be in a `saffron-rc2`.

Comment (nullpointer0x00) [reply app/upgrades.go]: Yeah, this isn't valid anymore.  I had this PR in draft mode until @iramiller updated me on what needed to happen.

File: app/navs_mainnet.go [hunk 5]
```diff
@@ -2,6 +2,7 @@ package app
 
 import (
 	"github.com/cosmos/cosmos-sdk/types"
+
 	markertypes "github.com/provenance-io/provenance/x/marker/types"
 )
```
Comment (nullpointer0x00) [inline app/navs_mainnet.go:7]: Note: This file was merged to main on accident.  Therefore, the diff is not showing up.  Must review this: https://github.com/provenance-io/provenance/blob/64e9cb994fb0d950c7b9a5c5ff32d8155c875cd6/app/navs_mainnet.go

File: app/navs_mainnet.go [hunk 6]
```diff
@@ -2,6 +2,7 @@ package app
...
```
Comment (iramiller) [inline app/navs_mainnet.go:8]: We should add a comment here (and/or rename this function) to indicates these markers are part of `pio-mainnet-1` and can't be ran against other networks. ... also should add the effective block height / date time these were created against as a comment too.

Comment (nullpointer0x00) [reply app/navs_mainnet.go:8]: Good call.  I updated it with more detail.

**Review by Taztingo — APPROVED** [review]

**Review by iramiller — APPROVED** [review]

File: app/upgrades.go (hunk 4 above)
Comment (iramiller) [reply app/upgrades.go]: We can't do this in any of the RCx handlers because the data is network specific ...

**Pull request merged**